### DISCUSSION TITLE
[New Feature] Beacon modes.

### DISCUSSION THREAD

**Discussion by Lylaktech** [Phobos-developers/Phobos#955] [Ideas and feedback]
### Description:
Allow for beacons to have multiple modes, similar to that of MOBAs where you can hold the button down then move mouse around to signal attack/defend/retreat/Superweapon.

### Additional Files / Links:
This is an example from Heroes of the Storm:
![1](https://user-images.githubusercontent.com/68288553/129413180-038ff650-ebd0-4903-ba06-68a015c77e04.jpg)

Additional Note: if designing a UI like this is extremely hard or impossible, perhaps it could have a different hotkey for different beacon types instead?

**Comment by secsome** [comment]
This first needs further investigate into BeaconClass, disassembling is needed.

**Comment by bigsinger** [comment]
这个功能太棒了！！！ 支持！
